public/app/types/unified-alerting-dto.ts

```diff
@@ -213,7 +213,7 @@ export interface GrafanaRuleDefinition extends PostableGrafanaRuleDefinition {
   id?: string;
   uid: string;
   namespace_uid: string;
-  namespace_id: number;
+  namespace_id?: number; //deprecated
```

Comment:
You can mark it as deprecated with the following syntax :)
```suggestion
  /** @deprecated use namespace_uid instead */
  namespace_id: number;
```